idaes/generic_models/properties/core/generic/tests/test_electrolyte_components.py

```diff
@@ -0,0 +1,420 @@
+        "state_bounds": {"flow_mol": (0, 100, 1000, pyunits.mol/pyunits.s),
+                         "temperature": (273.15, 300, 500, pyunits.K),
+                         "pressure": (5e4, 1e5, 1e6, pyunits.Pa)},
+        "state_components": StateIndex.true,
+        "pressure_ref": (101325, pyunits.Pa),
+        "temperature_ref": (298.15, pyunits.K),
+
+        # Defining phase equilibria
+        }
+
+    @pytest.mark.unit
+    def test_true_component_lists_2_phase(self):
+        m = ConcreteModel()
+
+        m.fs = FlowsheetBlock(default={'dynamic': False})
+
+        m.fs.props = GenericParameterBlock(
+            default=TestNonAqueousComponents.config)
+
+        m.fs.state = m.fs.props.build_state_block(
+                [1],
+                default={"defined_state": True})
+
+        assert m.fs.props._electrolyte
+
+        assert m.fs.props.anion_set == ["HCO3-"]
+        assert m.fs.props.cation_set == ["K+"]
+        assert m.fs.props.solvent_set == ["H2O"]
+        assert m.fs.props.solute_set == ["CO2"]
+        assert m.fs.props._apparent_set == ["KHCO3"]
+        assert m.fs.props._non_aqueous_set == ["N2"]
+
+        assert m.fs.props.true_species_set == [
+            "HCO3-", "K+", "H2O", "CO2", "N2"]
+        assert m.fs.props.apparent_species_set == [
+            "H2O", "CO2", "KHCO3", "N2"]
+        assert m.fs.props.component_list == [
+            "HCO3-", "K+", "H2O", "CO2", "KHCO3", "N2"]
+
+        assert m.fs.props.true_phase_component_set == [
+            ("Liq", "HCO3-"), ("Liq", "K+"), ("Liq", "H2O"), ("Liq", "CO2"),
+            ("Vap", "H2O"), ("Vap", "CO2"), ("Vap", "N2")]
+        assert m.fs.props.apparent_phase_component_set == [
+            ("Liq", "H2O"), ("Liq", "CO2"), ("Liq", "KHCO3"),
+            ("Vap", "H2O"), ("Vap", "CO2"), ("Vap", "KHCO3"), ("Vap", "N2")]
+
+        assert m.fs.state[1].component_list is m.fs.props.true_species_set
+
+        assert m.fs.state[1].phase_component_set is \
+            m.fs.props.true_phase_component_set
+
+        assert isinstance(m.fs.state[1].flow_mol, Var)
+        assert len(m.fs.state[1].flow_mol) == 1
+        assert isinstance(m.fs.state[1].pressure, Var)
+        assert len(m.fs.state[1].pressure) == 1
+        assert isinstance(m.fs.state[1].temperature, Var)
+        assert len(m.fs.state[1].temperature) == 1
+
+        assert isinstance(m.fs.state[1].flow_mol_phase, Var)
+        assert len(m.fs.state[1].flow_mol_phase) == 2
+        for p in m.fs.state[1].flow_mol_phase:
+            assert p in ["Liq", "Vap"]
+        assert isinstance(m.fs.state[1].phase_frac, Var)
+        assert len(m.fs.state[1].phase_frac) == 2
+        for p in m.fs.state[1].phase_frac:
+            assert p in ["Liq", "Vap"]
+
+        assert isinstance(m.fs.state[1].mole_frac_comp, Var)
+        assert len(m.fs.state[1].mole_frac_comp) == 5
+        for j in m.fs.state[1].mole_frac_comp:
+            assert j in ["HCO3-", "K+", "H2O", "CO2", "N2"]
```

Comment:
Ok, so the state block does not explicitly include apparent species, right? How will users access the apparent species?